src/main/java/org/jabref/logic/git/GitClientHandler.java

```diff
@@ -0,0 +1,139 @@
+package org.jabref.logic.git;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Optional;
+
+import org.jabref.gui.DialogService;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.preferences.AutoPushMode;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.ResetCommand;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.lib.Ref;
+import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider;
+
+public class GitClientHandler extends GitHandler {
+    private final DialogService dialogService;
+    private final GitPreferences gitPreferences;
+
+    public GitClientHandler(Path repositoryPath,
+                            DialogService dialogService,
+                            GitPreferences gitPreferences) {
+        super(repositoryPath, false);
+        this.dialogService = dialogService;
+        this.gitPreferences = gitPreferences;
+
+        this.credentialsProvider = new UsernamePasswordCredentialsProvider(
+                gitPreferences.getGitHubUsername(),
+                gitPreferences.getGitHubPasskey()
+        );
+    }
+
+    /**
+     * Contains logic for commiting and pushing after a database is saved locally,
+     * if the relevant preferences are present.<p>
+     * A git commit is created and a 'git pull --rebase' is executed. In the case of
```

Comment:
Reading this and the code - it feels complicated. Maybe in the last years, I got too excited about git features. Can you remove the logic for `pull --rebase` and just do a `git pull`?